### PR TITLE
fix: datacatalog taxonomy import project from name

### PR DESCRIPTION
**Pull request by ekapratama93**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/13307

This is re-attempt to merge #7063 with small change.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
datacatalog: fixed the import failure when the `project` is different from the default on `google_data_catalog_taxonomy`
```

### CODE REVIEW

**Comment by ekapratama93** [comment]
Running acceptance test for `TestAccRegionInstanceGroupManager_stateful` in my local environment was successful on this changes.

```sh
make testacc TEST=./google-beta TESTARGS='-run=TestAccRegionInstanceGroupManager_stateful'
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccRegionInstanceGroupManager_stateful -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccRegionInstanceGroupManager_stateful
=== PAUSE TestAccRegionInstanceGroupManager_stateful
=== CONT  TestAccRegionInstanceGroupManager_stateful
--- PASS: TestAccRegionInstanceGroupManager_stateful (416.06s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 416.780s
```

**Comment by slevenick** [comment]
Requesting review from @melinath who reverted the previous PR

File: mmv1/third_party/terraform/tests/resource_compute_region_instance_group_manager_test.go.erb [hunk 1]
```diff
@@ -1405,7 +1405,7 @@ resource "google_compute_region_instance_group_manager" "igm-basic" {
     device_name = "stateful-disk"
     delete_rule = "NEVER"
   }
-  <% unless version == "ga" -%>
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_compute_region_instance_group_manager_test.go.erb:1408]: Could the changes to this file be unreverted? seems unrelated to the change.

**Review by melinath — CHANGES_REQUESTED** [review]
overall this LGTM - one change request below.

File: mmv1/third_party/terraform/tests/resource_compute_region_instance_group_manager_test.go.erb (hunk 1 above)
Comment (bakuljajan) [reply mmv1/third_party/terraform/tests/resource_compute_region_instance_group_manager_test.go.erb:1408]: yes sorry, I changed it when debugging some weird VCR test failure.

**Review by melinath — APPROVED** [review]
LGTM - this isn't currently caught by automation but I've confirmed the "region" field exists, and the tests aren't crashing, so it should be fine. The tests will start getting run as part of https://github.com/GoogleCloudPlatform/magic-modules/pull/7185 or https://github.com/GoogleCloudPlatform/magic-modules/pull/6580

**Comment by melinath** [comment]
failed test looks unrelated
